pkg/operator/controller/ingress/status.go

```diff
@@ -979,7 +979,14 @@ func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNS
 		}
 	}
 	var conditions []operatorv1.OperatorCondition
-	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
+	if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) && ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
+		conditions = append(conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.DNSManagedIngressConditionType,
+			Status:  operatorv1.ConditionFalse,
+			Reason:  "DomainNotMatching",
+			Message: "DNS management is not supported for ingresscontrollers with domain not matching the baseDomain of the cluster DNS config.",
+		})
+	} else if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
```

Comment:
```suggestion
	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
	    if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) {
	  	    conditions = append(conditions, operatorv1.OperatorCondition{
			Type:    operatorv1.DNSManagedIngressConditionType,
			Status:  operatorv1.ConditionFalse,
			Reason:  "DomainNotMatching",
			Message: fmt.Sprintf("Domain %s not matching baseDomain %s.",ic.Status.Domain, dnsConfig.Spec.BaseDomain),
		    })
	} else {
```

Reply:
```suggestion
	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
		if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) {
			conditions = append(conditions, operatorv1.OperatorCondition{
				Type:    operatorv1.DNSManagedIngressConditionType,
				Status:  operatorv1.ConditionFalse,
				Reason:  "DomainNotMatching",
				Message: "DNS management is not supported for ingresscontrollers with domain not matching the baseDomain of the cluster DNS config.",
			})
		} else {
			conditions = append(conditions, operatorv1.OperatorCondition{
				Type:    operatorv1.DNSManagedIngressConditionType,
				Status:  operatorv1.ConditionFalse,
				Reason:  "UnmanagedLoadBalancerDNS",
				Message: "The DNS management policy is set to Unmanaged.",
			})
		}
	} else {
		conditions = append(conditions, operatorv1.OperatorCondition{
			Type:    operatorv1.DNSManagedIngressConditionType,
			Status:  operatorv1.ConditionTrue,
			Reason:  "Normal",
			Message: "DNS management is supported and zones are specified in the cluster DNS config.",
		})
	}
```
Hmm, are you suggesting I change the `if..else..if` like this?

Reply:
You had `if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS ` in both places, and I was suggesting to put that in the outer `if` clause.